src/pages/docs/registration-authorities/acme-for-vault.mdx

```diff
@@ -0,0 +1,266 @@
+---
+title: ACME Registration Authority for Hashicorp Vault CA
+html_title: ACME Registration Authority for Hashicorp Vault CA
+description: How to configure a Smallstep ACME Registration Authority with Hashicorp Vault CA. 
+---
+
+
+A Registration Authority sits in front of a Certificate Authority (CA) and provides an authentication and authorization layer.
+This guide will show you how to set up an ACME Registration Authority (RA) using `step-ca`, backed by a [Hashicorp Vault Certificate Authority][hvca] (HVCA) instance.
+
+Note about Hashicorp Vault Certificate Authority:
+
+> Vault's PKI secrets engine can dynamically generate X.509 certificates on demand. This allows services to acquire certificates without going through the usual manual process of generating a private key and Certificate Signing Request (CSR), submitting to a Certificate Authority (CA), and then waiting for the verification and signing process to complete.
+
+[hvca (onprem)]: https://learn.hashicorp.com/tutorials/vault/pki-engine
+[hvca (cloud)]: https://cloud.hashicorp.com/products/vault
+
+## Requirements
+
+* [`step-ca`](https://smallstep.com/docs/step-ca/installation)
+* [Hashicorp Vault](https://learn.hashicorp.com/tutorials/vault/getting-started-deploy)
+* [`jq`](https://stedolan.github.io/jq/download/) tool to parse JSON output.
+
+When not specified, we run `vault` commands as `root`. A dedicated profile will be created after configuration.
+
+## Configuration Guide
+
+### 1. Start a Vault server and log into vault
+
+For testing, you can run `vault server -dev` in a terminal window, and use an HTTP Vault address as shown here.
+Note: in dev mode, you will lose your PKI when you exit the Vault server.
```

Comment:
If the vault server is configured with TLS, we will need TLS we might need to define the CA to use, client certificates, ... This can be achieved with the same environment variables CLI uses, see https://www.vaultproject.io/docs/commands#environment-variables

I haven't tested this.